tracing-syslog/src/syslog.rs

```diff
@@ -0,0 +1,220 @@
+    pub const LOG_NOWAIT: Self = Self(libc::LOG_NOWAIT);
+    /// Log to stderr as well.
+    pub const LOG_PERROR: Self = Self(libc::LOG_PERROR);
+}
+
+impl std::ops::BitOr for Options {
+    type Output = Self;
+    fn bitor(self, rhs: Self) -> Self::Output {
+        Self(self.0 | rhs.0)
+    }
+}
+
+/// `syslog` facility.
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone, Debug)]
+#[repr(i32)]
+pub enum Facility {
+    /// Kernel messages (these can't be generated from user processes).
+    LOG_KERN = libc::LOG_KERN,
+    /// Generic user-level messages.
+    LOG_USER = libc::LOG_USER,
+    /// Mail subsystem.
+    LOG_MAIL = libc::LOG_MAIL,
+    /// System daemons without separate facility value.
+    LOG_DAEMON = libc::LOG_DAEMON,
+    /// Security/authorization messages.
+    LOG_AUTH = libc::LOG_AUTH,
+    /// Messages generated internally by `syslogd(8)`.
+    LOG_SYSLOG = libc::LOG_SYSLOG,
+    /// Line printer subsystem.
+    LOG_LPR = libc::LOG_LPR,
+    /// USENET news subsystem.
+    LOG_NEWS = libc::LOG_NEWS,
+    /// UUCP subsystem.
+    LOG_UUCP = libc::LOG_UUCP,
+    /// Clock daemon (`cron` and `at`).
+    LOG_CRON = libc::LOG_CRON,
+    /// Security/authorization messages (private).
+    LOG_AUTHPRIV = libc::LOG_AUTHPRIV,
+    /// FTP daemon.
+    LOG_FTP = libc::LOG_FTP,
+    /// Reserved for local use.
+    LOG_LOCAL0 = libc::LOG_LOCAL0,
+    /// Reserved for local use.
+    LOG_LOCAL1 = libc::LOG_LOCAL1,
+    /// Reserved for local use.
+    LOG_LOCAL2 = libc::LOG_LOCAL2,
+    /// Reserved for local use.
+    LOG_LOCAL3 = libc::LOG_LOCAL3,
+    /// Reserved for local use.
+    LOG_LOCAL4 = libc::LOG_LOCAL4,
+    /// Reserved for local use.
+    LOG_LOCAL5 = libc::LOG_LOCAL5,
+    /// Reserved for local use.
+    LOG_LOCAL6 = libc::LOG_LOCAL6,
+    /// Reserved for local use.
+    LOG_LOCAL7 = libc::LOG_LOCAL7,
+}
+
+impl Default for Facility {
+    fn default() -> Self {
+        Self::LOG_USER
+    }
+}
+
+/// `syslog` severity.
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone)]
+#[repr(i32)]
+pub enum Severity {
+    /// System is unusable.
+    LOG_EMERG = libc::LOG_EMERG,
+    /// Action must be taken immediately.
+    LOG_ALERT = libc::LOG_ALERT,
+    /// Critical conditions.
+    LOG_CRIT = libc::LOG_CRIT,
+    /// Error conditions.
+    LOG_ERR = libc::LOG_ERR,
+    /// Warning conditions.
+    LOG_WARNING = libc::LOG_WARNING,
+    /// Normal, but significant, condition.
+    LOG_NOTICE = libc::LOG_NOTICE,
+    /// Informational message.
+    LOG_INFO = libc::LOG_INFO,
+    /// Debug-level message.
+    LOG_DEBUG = libc::LOG_DEBUG,
+}
+
+impl From<Level> for Severity {
+    fn from(level: Level) -> Self {
+        match level {
+            Level::ERROR => Self::LOG_ERR,
+            Level::WARN => Self::LOG_WARNING,
+            Level::INFO => Self::LOG_NOTICE,
+            Level::DEBUG => Self::LOG_INFO,
+            Level::TRACE => Self::LOG_DEBUG,
+        }
+    }
+}
```

Comment:
Why this conversion of levels? Can you add a comment explaining the reason for this mapping?

Reply:
I believe this is basically the same as [what we do in the `tracing-journald` crate](https://github.com/tokio-rs/tracing/blob/27b28773513696220d7dca0d800fe25da67b0f68/tracing-journald/src/lib.rs#L59-L67), which I suggested in https://github.com/tokio-rs/tracing/pull/691#issuecomment-620147089. syslog lacks a "trace" level, and the "notice" level is roughly equivalent to what we call "info". 

A comment might still be nice, but that's the backstory.

Reply:
Yup, I was basing it on `tracing-journald`, I'll document it. I could also add an option to specify a different mapping function.

Reply:
If we don't see the need for custom level mappers, I could make `Severity` private, which would also resolve the above comment about representations and breaking changes